### ISSUE TITLE
Display issues with mesa-21.2.3-6

### ISSUE THREAD

**Issue by joepvd** [qutebrowser/qutebrowser#6879]
<details>
<summary>Version info</summary>

```
qutebrowser v2.4.0
Backend: QtWebEngine 5.15.6, based on Chromium 87.0.4280.144
Qt: 5.15.2

CPython: 3.10.0
PyQt: 5.15.6

sip: 6.1.0.dev2104271705
colorama: 0.4.4
jinja2: 3.0.1
pygments: 2.9.0
yaml: 5.4.1
adblock: no
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebEngine: 5.15.2
PyQt5.QtWebKitWidgets: yes
pdf.js: no
sqlite: 3.36.0
QtNetwork SSL: OpenSSL 1.1.1l  FIPS 24 Aug 2021

Style: QFusionStyle
Platform plugin: xcb
OpenGL: Intel, 4.6 (Compatibility Profile) Mesa 21.2.3
Platform: Linux-5.15.6-200.fc35.x86_64-x86_64-with-glibc2.34, 64bit
Linux distribution: Fedora Linux 35 (Thirty Five) (fedora)
Frozen: False
Imported from /usr/lib/python3.10/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib64/qt5/libexec, data path: /usr/share/qt5

Paths:
cache: /home/jdelft/.cache/qutebrowser
config: /home/jdelft/.config/qutebrowser
data: /home/jdelft/.local/share/qutebrowser
runtime: /run/user/1000/qutebrowser

Autoconfig loaded: yes
Config.py: /home/jdelft/.config/qutebrowser/config.py has been loaded
Uptime: 1 day, 10:41:08
```
</details>

*Does the bug happen if you start with `--temp-basedir`?*
Yes

*Description*
When I upgrade my Fedora 35 installation to the current mesa, there is screen tearing and flashing on some websites. A solid reproducer is reddit (with dark theme and compact view). Parts of the site seem screen shotted and appear in the view, see screen shot. Also when hovering over items , the screen will very briefly flash white.

![2021-12-08_20-48](https://user-images.githubusercontent.com/884975/145549708-9cf4b7bc-4151-4e23-b85e-b04e87815392.png)

The affected mesa version is `21.2.3-6.fc35`. 
<details><summary>All packages that change with an upgrade</summary>

```
(1/20): clang-13.0.0-3.fc35.x86_64.rpm                                                                                                                      559 kB/s |  84 kB     00:00
(2/20): clang-resource-filesystem-13.0.0-3.fc35.x86_64.rpm                                                                                                  212 kB/s |  14 kB     00:00
(3/20): clang-devel-13.0.0-3.fc35.x86_64.rpm                                                                                                                3.1 MB/s | 2.2 MB     00:00
(4/20): compiler-rt-13.0.0-1.fc35.x86_64.rpm                                                                                                                3.1 MB/s | 2.4 MB     00:00
(5/20): libomp-devel-13.0.0-1.fc35.x86_64.rpm                                                                                                               833 kB/s |  29 kB     00:00
(6/20): clang-libs-13.0.0-3.fc35.x86_64.rpm                                                                                                                 5.1 MB/s |  22 MB     00:04
(7/20): llvm-13.0.0-4.fc35.x86_64.rpm                                                                                                                       3.1 MB/s |  15 MB     00:04
(8/20): clang-tools-extra-13.0.0-3.fc35.x86_64.rpm                                                                                                          2.3 MB/s |  14 MB     00:06
(9/20): mesa-filesystem-21.3.1-2.fc35.x86_64.rpm                                                                                                            517 kB/s |  18 kB     00:00
(10/20): mesa-libEGL-21.3.1-2.fc35.x86_64.rpm                                                                                                               1.5 MB/s | 121 kB     00:00
(11/20): mesa-libEGL-devel-21.3.1-2.fc35.x86_64.rpm                                                                                                         436 kB/s |  18 kB     00:00
(12/20): mesa-libGL-21.3.1-2.fc35.x86_64.rpm                                                                                                                1.7 MB/s | 170 kB     00:00
(13/20): mesa-libGL-devel-21.3.1-2.fc35.x86_64.rpm                                                                                                          520 kB/s |  35 kB     00:00
(14/20): mesa-libOpenCL-21.3.1-2.fc35.x86_64.rpm                                                                                                            1.8 MB/s | 579 kB     00:00
(15/20): mesa-libgbm-21.3.1-2.fc35.x86_64.rpm                                                                                                               493 kB/s |  44 kB     00:00
(16/20): mesa-libglapi-21.3.1-2.fc35.x86_64.rpm                                                                                                             567 kB/s |  55 kB     00:00
(17/20): mesa-libxatracker-21.3.1-2.fc35.x86_64.rpm                                                                                                         2.1 MB/s | 2.0 MB     00:00
(18/20): llvm-libs-13.0.0-4.fc35.x86_64.rpm                                                                                                                 5.0 MB/s |  25 MB     00:04
(19/20): mesa-vulkan-drivers-21.3.1-2.fc35.x86_64.rpm                                                                                                       3.0 MB/s | 5.6 MB     00:01
(20/20): mesa-dri-drivers-21.3.1-2.fc35.x86_64.rpm                                                                                                          5.3 MB/s |  26 MB     00:04
```

</details>

*How to reproduce*
```
sudo dnf upgrade -y && killall qutebrowser && qutebrowser --temp-basedir https://reddit.com/r/qutebrowser
```
Then set to Dark Made and Compact View, and hover over the posts.

After downgrading `mesa-*`, it is fine again. These are the packages and versions involved:

<details><summary>Package change with downgrade</summary>

```
⫸  sudo dnf downgrade 'mesa-*'
Last metadata expiration check: 0:49:18 ago on Fri 10 Dec 2021 08:58:24 AM CET.
Package mesa-libGLU of lowest version already installed, cannot downgrade it.
Dependencies resolved.
============================================================================================================================================================================================
 Package                                                 Architecture                         Version                                            Repository                            Size
============================================================================================================================================================================================
Downgrading:
 clang                                                   x86_64                               13.0.0~rc1-1.fc35                                  fedora                                83 k
 clang-devel                                             x86_64                               13.0.0~rc1-1.fc35                                  fedora                               2.2 M
 clang-libs                                              x86_64                               13.0.0~rc1-1.fc35                                  fedora                                22 M
 clang-resource-filesystem                               x86_64                               13.0.0~rc1-1.fc35                                  fedora                                14 k
 clang-tools-extra                                       x86_64                               13.0.0~rc1-1.fc35                                  fedora                                14 M
 compiler-rt                                             x86_64                               13.0.0~rc1-1.fc35                                  fedora                               2.4 M
 libomp-devel                                            x86_64                               13.0.0~rc1-1.fc35                                  fedora                                29 k
 llvm                                                    x86_64                               13.0.0~rc1-1.fc35                                  fedora                                15 M
 llvm-libs                                               x86_64                               13.0.0~rc1-1.fc35                                  fedora                                25 M
 mesa-dri-drivers                                        x86_64                               21.2.3-6.fc35                                      fedora                                24 M
 mesa-filesystem                                         x86_64                               21.2.3-6.fc35                                      fedora                                18 k
 mesa-libEGL                                             x86_64                               21.2.3-6.fc35                                      fedora                               120 k
 mesa-libEGL-devel                                       x86_64                               21.2.3-6.fc35                                      fedora                                18 k
 mesa-libGL                                              x86_64                               21.2.3-6.fc35                                      fedora                               170 k
 mesa-libGL-devel                                        x86_64                               21.2.3-6.fc35                                      fedora                                35 k
 mesa-libOpenCL                                          x86_64                               21.2.3-6.fc35                                      fedora                               556 k
 mesa-libgbm                                             x86_64                               21.2.3-6.fc35                                      fedora                                43 k
 mesa-libglapi                                           x86_64                               21.2.3-6.fc35                                      fedora                                53 k
 mesa-libxatracker                                       x86_64                               21.2.3-6.fc35                                      fedora                               1.9 M
 mesa-vulkan-drivers                                     x86_64                               21.2.3-6.fc35                                      fedora                               5.0 M
```
</details>

This probably is a bug in mesa or qt or whereever, but for me it only shows in qutebrowser, which makes it seem the right place to start digging, yet I'm out of my depth. Setting `MESA_DEBUG` does not show anything interesting.

I am on i3wm, Xorg, no compositor on a Lenovo T580 with an intel graphical card.

**Comment by The-Compiler** [comment]
I doubt there's anything qutebrowser can do about this. Could you try with another project using QtWebEngine, such as KDE Falkon?

**Comment by joepvd** [comment]
Wow how can you reply within 20 seconds? 🚀 

Took Falkon for a spin. The artifacts do not show, yet the occasional flicker is there.

**Comment by joepvd** [comment]
Tagged on to https://gitlab.freedesktop.org/mesa/mesa/-/issues/5731. Closing here.

Thanks again!

**Comment by prosoitos** [comment]
For info, I opened a new issue on mesa with v. 21.3.1: https://gitlab.freedesktop.org/mesa/mesa/-/issues/5745

**Comment by The-Compiler** [comment]
The [fix](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/14783/diffs?commit_id=180c012807d2799c4366abb9b2c24001a40347a0) doesn't seem to have made it into a release upstream still. However, Archlinux [added the patch](https://github.com/archlinux/svntogit-packages/commit/854c8aeaac5691ec04b97f762ddf2489e9767ab8#diff-3e341d2d9c67be01819b25b25d5e53ea3cdf3a38d28846cda85a195eb9b7203a) to their 21.3.5 package.

**Comment by joepvd** [comment]
Fedora has the fix incorporated since around two weeks with [mesa-21.3.4-3.fc35](https://bodhi.fedoraproject.org/updates/FEDORA-2022-d9c9d567ce).

**Comment by The-Compiler** [comment]
Looks like this has finally been fixed upstream with [mesa 21.3.6](https://docs.mesa3d.org/relnotes/21.3.6.html#bug-fixes).
